### PR TITLE
One-Time-Checkout: Cover transaction cost

### PR DESCRIPTION
**Pull request by paul-daniel-dempsey**
## What are you doing in this PR?
Add cover transaction cost to generic one-time-checkout.

[**Trello Card**](https://trello.com/c/beKF97Yg/1195-new-one-time-implement-cover-transaction-element)

|before|after|
|------|------|
|image|image|

### CODE REVIEW

File: support-frontend/assets/pages/[countryGroupId]/oneTimeCheckout.tsx [hunk 1]
```diff
@@ -282,7 +283,23 @@ function OneTimeCheckoutComponent({
...
 
 	const [otherAmountError, setOtherAmountError] = useState<string>();
-	const finalAmount = getFinalAmount(selectedPriceCard, otherAmount, minAmount);
+	const [coverTransactionCost, setCoverTransactionCost] =
+		useState<boolean>(false);
+
+	const amountWithoutCoverCost =
+		getFinalAmount(selectedPriceCard, otherAmount, minAmount, false) ?? 0;
+	const transactionCoverCost = amountWithoutCoverCost * 0.04;
+	const transactionCostCopy = `I’d like to add a further ${simpleFormatAmount(
```
Comment (andrewHEguardian) [inline support-frontend/assets/pages/[countryGroupId]/oneTimeCheckout.tsx:292]: We could put the copy inside the component itself

File: support-frontend/assets/pages/[countryGroupId]/oneTimeCheckout.tsx [hunk 2]
```diff
@@ -831,6 +848,17 @@ function OneTimeCheckoutComponent({
...
 						</FormSection>
+						<CoverTransactionCost
+							transactionCost={coverTransactionCost}
+							transactionCostCopy={transactionCostCopy}
+							onChecked={(check) => {
+								setCoverTransactionCost(check);
+							}}
+							transactionCostAmount={simpleFormatAmount(
+								currency,
+								finalAmount ? finalAmount : 0,
```
Comment (andrewHEguardian) [inline support-frontend/assets/pages/[countryGroupId]/oneTimeCheckout.tsx:859]: Is showing "I’d like to add a further £0" the copy we want? I guess it's an edge case

File: support-frontend/assets/helpers/forms/checkouts.ts [hunk 3]
```diff
@@ -341,6 +341,7 @@ function getAvailablePaymentRequestButtonPaymentMethod(
 
 // ----- Exports ----- //
 export {
+	round,
```
Comment (andrewHEguardian) [inline support-frontend/assets/helpers/forms/checkouts.ts:344]: Why do we need a new export? Can't we use the same rounding function as the old checkout?

**Review by andrewHEguardian — APPROVED** [review]
Neat! 👍  Just a couple of small questions

File: support-frontend/assets/pages/[countryGroupId]/oneTimeCheckout.tsx (hunk 2 above)
Comment (paul-daniel-dempsey) [reply support-frontend/assets/pages/[countryGroupId]/oneTimeCheckout.tsx:859]: Yep, I checked that and the current checkout does this, so presumed was ok.
image

File: support-frontend/assets/helpers/forms/checkouts.ts (hunk 3 above)
Comment (paul-daniel-dempsey) [reply support-frontend/assets/helpers/forms/checkouts.ts:344]: Hmmm, this has been a thorny problem this afternoon! The old checkout does not multiply the price by 1.04 in the same way its done in the one-time-checkout. Oddly am getting something like `60 x 1.04 = 62.40000000001` hence using the round. Will spend a bit more time looking into a workaround without it now.

File: support-frontend/assets/pages/[countryGroupId]/oneTimeCheckout.tsx (hunk 1 above)
Comment (paul-daniel-dempsey) [reply support-frontend/assets/pages/[countryGroupId]/oneTimeCheckout.tsx]: Yep good idea, have moved it inside.

File: support-frontend/assets/helpers/forms/checkouts.ts (hunk 3 above)
Comment (paul-daniel-dempsey) [reply support-frontend/assets/helpers/forms/checkouts.ts:344]: Ok have used `roundToDecimalPlaces` as used in old checkout, not needing to expose `round`

**Pull request merged**